virtool/references/db.py

```diff
@@ -239,64 +240,82 @@ def __init__(self, app, task_id):
+        self.non_existent_references = [ref_id for ref_id in reference_ids if ref_id not in existent_references]
 
-    async def remove_unreferenced_otus(self):
-        ref_id = self.context["ref_id"]
+        for dir_name in self.non_existent_references:
+            shutil.rmtree(os.path.join(path, dir_name))
 
-        referenced_otu_ids = await self.db.analyses.distinct("results.otu.id", {"reference.id": ref_id})
+    async def remove_indexes(self):
+        for ref_id in self.non_existent_references:
+            await self.db.indexes.delete_many({
+                "reference.id": ref_id
+            })
 
-        unreferenced_otu_ids = await self.db.otus.distinct("_id", {
-            "reference.id": ref_id,
-            "_id": {
-                "$not": {
-                    "$in": referenced_otu_ids
+    async def remove_unreferenced_otus(self):
+        for ref_id in self.non_existent_references:
+            referenced_otu_ids = await self.db.analyses.distinct("results.otu.id", {"reference.id": ref_id})
+
+            unreferenced_otu_ids = await self.db.otus.distinct("_id", {
+                "reference.id": ref_id,
+                "_id": {
+                    "$not": {
+                        "$in": referenced_otu_ids
+                    }
                 }
-            }
-        })
+            })
 
-        diff_file_change_ids = await self.db.history.distinct("_id", {
-            "diff": "file",
-            "otu.id": {
-                "$in": unreferenced_otu_ids
-            }
-        })
+            diff_file_change_ids = await self.db.history.distinct("_id", {
+                "diff": "file",
+                "otu.id": {
+                    "$in": unreferenced_otu_ids
+                }
+            })
 
-        await asyncio.gather(
-            self.db.otus.delete_many({"_id": {"$in": unreferenced_otu_ids}}),
-            self.db.history.delete_many({"otu.id": {"$in": unreferenced_otu_ids}}),
-            self.db.sequences.delete_many({"otu_id": {"$in": unreferenced_otu_ids}}),
-            virtool.history.utils.remove_diff_files(self.app, diff_file_change_ids)
-        )
+            await asyncio.gather(
+                self.db.otus.delete_many({"_id": {"$in": unreferenced_otu_ids}}),
+                self.db.history.delete_many({"otu.id": {"$in": unreferenced_otu_ids}}),
+                self.db.sequences.delete_many({"otu_id": {"$in": unreferenced_otu_ids}}),
+                virtool.history.utils.remove_diff_files(self.app, diff_file_change_ids)
+            )
 
     async def remove_referenced_otus(self):
-        ref_id = self.context["ref_id"]
         user_id = self.context["user_id"]
 
-        otu_count = await self.db.otus.count_documents({"reference.id": ref_id})
+        for ref_id in self.non_existent_references:
+            otu_count = await self.db.otus.count_documents({"reference.id": ref_id})
 
-        tracker = self.get_tracker(otu_count)
+            tracker = self.get_tracker(otu_count)
```

Comment:
Remove use of `tracker`. It adds complexity and the progress is not displayed to the users for this task anyway.